crates/wasi-provider/src/lib.rs

```diff
@@ -107,18 +113,48 @@ impl<S: ModuleStore + Send + Sync> Provider for WasiProvider<S> {
 
         let mut modules = self.store.fetch_pod_modules(&pod).await?;
         let client = kube::Client::from(self.kubeconfig.clone());
+        // TODO: Make volume refs part of the handle so they don't get dropped
+        let volumes = VolumeRef::volumes_from_pod(&self.volume_path, &pod, &client).await?;
         info!("Starting containers for pod {:?}", pod_name);
         for container in pod.containers() {
             let env = Self::env_vars(&container, &pod, &client).await;
             let module_data = modules
                 .remove(&container.name)
                 .expect("FATAL ERROR: module map not properly populated");
+            let container_volumes: HashMap<String, Option<String>> =
```

Comment:
Is there a reason we decided to make these Strings instead of `PathBufs`. It seems like a good idea to keep these `PathBuf`s for as long as we can

Reply:
So I thought they had to be strings when passing them down the line to the actual wasmtime setup and it turns out I remembered wrong. I'll refactor that